gym_solo/core/test_obs_observations.py

```diff
@@ -66,5 +66,157 @@ def test_compute(self, mock_vel, mock_base):
+        (0.19, -0.1046, 0.0), (0.05997269288895365, 0.0, 0.0, 0.998200018086379), -1), 
+    (4, b'FR_KFE', 0, 10, 9, 1, 0.0, 0.0, -10.0, 10.0, 1000.0, 
+      1000.0, b'FR_LOWER_LEG', (0.0, 0.9996672990385953, 0.025793240061682075), 
+      (1.377e-05, -0.00822816082925067, -0.08287430545789741), 
+      (-0.04709322720802768, 0.0, 0.0, 0.9988904984787538), 3), 
+    (5, b'FR_ANKLE', 4, -1, -1, 0, 0.0, 0.0, -10.0, 10.0, 1000.0, 
+      1000.0, b'FR_FOOT', (0.0, 0.0, 0.0), (0.0, -0.014047115411452295, 
+      -0.07110382693156142), (-0.012897692844160934, 0.0, 0.0, 0.9999168213003008), 4), 
+    (6, b'HL_HFE', 0, 11, 10, 1, 0.0, 0.0, -10.0, 10.0, 1000.0, 
+      1000.0, b'HL_UPPER_LEG', (0.0, 0.9928065522152947, -0.11972948625288478), 
+      (-0.19, 0.1046, 0.0), (-0.05997269288895365, 0.0, 0.0, 0.998200018086379), -1), 
+    (7, b'HL_KFE', 0, 12, 11, 1, 0.0, 0.0, -10.0, 10.0, 1000.0, 
+      1000.0, b'HL_LOWER_LEG', (0.0, 0.9996672990385953, -0.025793240061682075), 
+      (-1.377e-05, 0.00822816082925067, -0.08287430545789741), 
+      (0.04709322720802768, 0.0, 0.0, 0.9988904984787538), 6), 
+    (8, b'HL_ANKLE', 4, -1, -1, 0, 0.0, 0.0, -10.0, 10.0, 1000.0, 
+      1000.0, b'HL_FOOT', (0.0, 0.0, 0.0), (0.0, -0.0017005235902268143, 
+      -0.07069750911605854), (0.012897692844160934, 0.0, 0.0, 0.9999168213003008), 7), 
+    (9, b'HR_HFE', 0, 13, 12, 1, 0.0, 0.0, -10.0, 10.0, 1000.0, 
+      1000.0, b'HR_UPPER_LEG', (0.0, 0.9928065522152947, 0.11972948625288478), 
+      (-0.19, -0.1046, 0.0), (0.05997269288895365, 0.0, 0.0, 0.998200018086379), -1), 
+    (10, b'HR_KFE', 0, 14, 13, 1, 0.0, 0.0, -10.0, 10.0, 1000.0, 
+      1000.0, b'HR_LOWER_LEG', (0.0, 0.9996672990385953, 0.025793240061682075), 
+      (1.377e-05, -0.00822816082925067, -0.08287430545789741), 
+      (-0.04709322720802768, 0.0, 0.0, 0.9988904984787538), 9), 
+    (11, b'HR_ANKLE', 4, -1, -1, 0, 0.0, 0.0, -10.0, 10.0, 1000.0, 
+      1000.0, b'HR_FOOT', (0.0, 0.0, 0.0), (0.0, -0.014047115411452295, 
+      -0.07110382693156142), (-0.012897692844160934, 0.0, 0.0, 0.9999168213003008), 10)]
+
+
+  @parameterized.expand([
+    ("default", False),
+    ("degrees", True)
+  ])
+  @mock.patch('pybullet.getNumJoints', autospec=True)
+  def test_attributes(self, name, degrees, mock_num_joints):
+    num_joints = 12
+    dummy_robot_id = 0
+    mock_num_joints.return_value = num_joints
+
+    o = obs.MotorEncoder(dummy_robot_id, degrees= degrees)
+    
+    self.assertEqual(o.robot, dummy_robot_id)
+    self.assertEqual(o._degrees, degrees)
+    self.assertEqual(o.num_joints, num_joints)
+
+  @parameterized.expand([
+    ("default", False),
+    ("degrees", True)
+  ])
+  @mock.patch('pybullet.getJointInfo', autospec=True)
+  @mock.patch('pybullet.getNumJoints', autospec=True)
+  def test_observation_space(self, name, degrees, mock_num_joints, mock_joint_info):
+    
+    def joint_info_side_effect(robot, joint):
+      return self.joint_info[joint]
```

Comment:
nit: make into a `lambda` function

Reply:
or make it a module function if you're gonna use in multiple places

Reply:
Will probably make it a lambda